minio/operator
# Remove hardcoded storage class

Remove hardcoded storage class to allow usage of the installed, default storage class resource, and not force the usage of the `standard` storage class which may not exist on a particular k8s deployment. 
The default storage class is automatically chosen by k8s on deployment of the minio tenant.


### Test steps (using an lxc instance)
Note: The desired result is that basically the guide at https://github.com/minio/operator/tree/master/helm/operator or https://min.io/docs/minio/kubernetes/upstream/operations/install-deploy-manage/deploy-minio-tenant-helm.html#deploy-tenant-helm should work out of the box with no failures. 


#### Persist the session even after rebooting. This is needed only on lxc instances.
```
loginctl enable-linger ubuntu
```

#### Install k3s. Print versions.
```
sudo touch /dev/kmsg
curl -sfL https://get.k3s.io | K3S_KUBECONFIG_MODE="644" sh -s - --snapshotter=fuse-overlayfs
export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
k3s --version
```
Output
```
k3s version v1.27.7+k3s2 (575bce76)
go version go1.20.10
```
```
sudo systemctl status k3s.service
```
```
● k3s.service - Lightweight Kubernetes
     Loaded: loaded (/etc/systemd/system/k3s.service; enabled; preset: enabled)
     Active: active (running) since Thu 2023-10-12 19:00:40 UTC; 26min ago
```

#### Install helm
```
curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3
chmod 700 get_helm.sh
./get_helm.sh
helm version
```
```
version.BuildInfo{Version:"v3.13.1", GitCommit:"3547a4b5bf5edb5478ce352e18858d8a552a4110", GitTreeState:"clean", GoVersion:"go1.20.8"}
```

#### Install minio-operator. Print version.
```
helm repo add minio https://operator.min.io/
helm install \
  --namespace minio-operator \
  --create-namespace minio-operator minio/operator
```

#### Validate minio-operator running
```
kubectl get pods -n minio-operator
```
```
NAME                              READY   STATUS    RESTARTS   AGE
minio-operator-7d988dd6b6-lckt4   1/1     Running   0          24m
console-5756b69c9f-8s8d8          1/1     Running   0          24m
minio-operator-7d988dd6b6-swvnj   1/1     Running   0          24m
```

#### Add http Nodeport to service
```
kubectl patch service -n minio-operator console --type='merge' -p '{"spec":{"type": "NodePort", "ports":[{"name": "http", "port": 9090, "nodePort": 31000}]}}'
```

#### Get jwt and login e.g. http://operator-1605.minio.training:31000
```
kubectl -n minio-operator get secret console-sa-secret -o jsonpath="{.data.token}" | base64 --decode
```

#### Install tenant. Validate pods work. Note in this case the tenant was also edited to remove TLS for ease of testing
```
helm install --namespace tenant-ns --create-namespace tenant minio/tenant
kubectl --namespace tenant-ns get pods 
```

#### Add nodeport to access tenant
```
kubectl patch service -n tenant-ns myminio-console -p '{"spec":{"type": "NodePort", "ports":[{"name": "http-console", "port": 9090, "protocol": "TCP", "targetPort": 9090, "nodePort": 31090}]}}'
```

#### Using the default user/password, login e.g. http://operator-1605.minio.training:31090
